impeller/entity/geometry/superellipse_geometry.cc

```diff
@@ -0,0 +1,111 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <vector>
+
+#include "flutter/impeller/entity/geometry/superellipse_geometry.h"
+
+#include "impeller/geometry/constants.h"
+
+namespace impeller {
+
+SuperellipseGeometry::SuperellipseGeometry(const Point& center,
+                                           Scalar radius,
+                                           int degree,
+                                           Scalar alpha,
+                                           Scalar beta)
+    : center_(center),
+      degree_(degree),
+      radius_(radius),
+      alpha_(alpha),
+      beta_(beta) {}
+
+GeometryResult SuperellipseGeometry::GetPositionBuffer(
+    const ContentContext& renderer,
+    const Entity& entity,
+    RenderPass& pass) const {
+  // https://math.stackexchange.com/questions/2573746/superellipse-parametric-equation
+  Scalar a = alpha_;
+  Scalar b = beta_;
+  Scalar n = degree_;
+
+  // TODO(jonahwilliams): determine parameter values based on scaling factor.
+  Scalar step = kPi / 80;
```

Comment:
Putting this part in the Tessellator would allow it to leverage the "GetTrigsForDivisions" caching.

Reply:
I'm going to leave this out for now so its stays separate before we decide how to ship it, but I agree it should eventually live in the tessellator.

Reply:
Good call if it isolates it until we figure out its future. It will matter a little for benchmarks, but not for testing its usefulness.